assemble.py

```diff
@@ -19,6 +19,50 @@
 logger = logging.getLogger("System Image Assembler")
 
 
+def remount_dev():
+    # containers don't see changes to /dev, so we have to hack around
+    # this by basically mounting a new /dev. The idea was inspired by
+    # this comment:
+    #  https://github.com/moby/moby/issues/27886#issuecomment-257244027
+    if getattr(remount_dev, 'called', None):
+        # you can't call remount_dev dev a 2nd time without a umount, but
+        # it also sometimes *fails* in between doing multiple target assemblies.
+        # My guess is its something with dockerd getting spawned and stopped.
+        # So just *try* to umount and hope for the best.
+        subprocess.call(["umount", "/dev"])
+    cmd('mount', '-t', 'devtmpfs', 'devtmpfs', '/dev')
+    setattr(remount_dev, 'called', True)
+
+
+def losetup(path: str) -> str:
+    # losetup can be tricky to run in a container. Due to the issues with
+    # /dev noted about in `remount_dev`, we periodically will have losetup
+    # run like:
+    #   bash-5.1# losetup -f /file
+    #   losetup: /file: No such file or directory
+    # However, losetup is giving misleading error message. strace will show the
+    # real issue is that losetup creates new loop device, say loop2. However,
+    # there's some kind of timing issue where /dev/loop2 doesn't always show
+    # up, thus the "No sush file or directory". When this happens, we need to
+    # remount /dev in order to see the new device.
+    try:
+        cmd('losetup', '-P', '-f', path)
```

Comment:
`kpartx -a -v path` can do the same job and maybe they don't have the same timing issue of losetup.
https://linux.die.net/man/8/kpartx
I don't do any test and my comment is only a suggestion.

Reply:
I am not sure this is related to timing, it rather has to do with dockerd/containerd ability to propagate changes from a host to container. If `losetup` is invoked in container changes has to made on a host to create a new lo device, and then this new device should become visible within container (that's my understanding).

Reply:
understand, so maybe `kpartx` will suffer from the same problem of the `losetup` as the container<=>host propagation path will be the same

Reply:
On the other side, `kpartx` is a modern tool and may use a more reliable way of creating/checking loop devices than losetup.
We could check it.

Reply:
kpartx looks interesting. I tried to google "kpartx vs losetup" to find some differences and don't get much. I say for now we merge this as a small improvement. and we can revisit kpartx in the future if we continue to have sporadic issues.